tests/functionnal/KnowbaseItem.php

```diff
@@ -343,41 +343,346 @@ public function testGetForCategory()
          ->contains(-1);
     }
 
+    protected function testComputeBooleanFullTextSearchProvider(): array
```

Comment:
Many cases should be added to validate all legitimate regex replacements and verify that there is no illegitimate replacements:
 - tests with spacing chars located at many places*
 - tests with quotes/asterisks located at many places*
 - tests with odd count of quotes
 - tests with parenthesis located at many places*
 - tests with odd count of parenthesis
 - tests with unexpected chars (neither letters, numbers, operators, ...)
 - tests with unicode chars (e.g. a search with cyrillic words)

*located at many places = at begin/end of search terms, inside words, surrounded by spaces, ...